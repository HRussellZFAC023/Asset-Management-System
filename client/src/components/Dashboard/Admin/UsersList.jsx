import React from "react";
import PropTypes from "prop-types";

const UsersList = props => {
  return (
    <div>
      {props.users.map(user => {
        return (
          <div key={user.id} className="box">
            <div className="columns is-vcentered">
              <div className="column is-9">
                <span><b>Username: </b>{user.username}</span>
                <br />
                <span><b>Email: </b>{user.email}</span>
              </div>
              <div className="column is-3">
                <button className="button is-danger is-fullwidth" onClick={() => props.delete(user.id)}>
                  <span>Delete</span>
                  <span className="icon is-small">
                    <i className="fa fa-times" />
                  </span>
                </button>
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );
};

UsersList.propTypes = {
  users: PropTypes.array.isRequired,
  delete: PropTypes.func.isRequired
};

export default UsersList;
