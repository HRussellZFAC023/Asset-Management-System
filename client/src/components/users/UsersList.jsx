import React from "react";
import PropTypes from "prop-types";

const UsersList = props => {
  return (
    <div>
      {props.users.map(user => {
        return (
          <div key={user.id} className="box title is-4 username">
            <span>{'Username: '}
            <p className="username">{user.username}</p></span>
            <br/> {'Email: ' + user.email}
          </div>
        );
      })}
    </div>
  );
};

UsersList.propTypes = {
  users: PropTypes.array.isRequired
};

export default UsersList;
