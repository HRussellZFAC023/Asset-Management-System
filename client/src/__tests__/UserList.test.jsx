import React from "react";
import { render, cleanup } from "@testing-library/react";

<<<<<<< HEAD:client/src/__tests__/UserList.test.jsx
import UsersList from '../components/users/UsersList';
=======
import UsersList from "../UsersList";
>>>>>>> master:client/src/components/__tests__/UserList.test.jsx

afterEach(cleanup);

const users = [
  {
    email: "hermanmu@gmail.com",
    id: 1,
    username: "michael"
  },
  {
    email: "michael@mherman.org",
    id: 2,
    username: "michaelherman"
  }
];

<<<<<<< HEAD:client/src/__tests__/UserList.test.jsx
it('renders a username', () => {
    const { getByText } = render(<UsersList users={users}/>);
    expect(getByText('michael')).toHaveClass('username');
    expect(getByText('michaelherman')).toHaveClass('username');
=======
it("renders a username", () => {
  const { getByText } = render(<UsersList users={users} />);
  expect(getByText("michael")).toHaveClass("username");
  expect(getByText("michaelherman")).toHaveClass("username");
>>>>>>> master:client/src/components/__tests__/UserList.test.jsx
});

it("renders", () => {
  const { asFragment } = render(<UsersList users={users} />);
  expect(asFragment()).toMatchSnapshot();
});
