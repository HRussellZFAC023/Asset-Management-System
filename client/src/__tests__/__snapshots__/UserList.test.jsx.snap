// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="box title is-4 username"
    >
      <span>
        Username: 
        <p
          class="username"
        >
          michael
        </p>
      </span>
      <br />
       Email: hermanmu@gmail.com
    </div>
    <div
      class="box title is-4 username"
    >
      <span>
        Username: 
        <p
          class="username"
        >
          michaelherman
        </p>
      </span>
      <br />
       Email: michael@mherman.org
    </div>
  </div>
</DocumentFragment>
`;
